### ISSUE TITLE
Unpublished related content appearing in Preview Mode / Live mode & Are published when Pushing to S3 - 5.3x

### ISSUE THREAD

**Issue by craigWagner99** [dotCMS/core#19044]
Zendesk ticket: https://dotcms.zendesk.com/agent/tickets/102004

Unpublished related items are appearing in Preview/Live mode and are getting moved to Published when they are pushed to a static, S3 site.

1. Create Relationship between two content types.
2. Create Page Asset
3. Add VTL to display the parent content and loop through the child content int he relationship.
4. Publish page.
5. Preview content in Live mode. 
...
6. Push Publish to a static S3 endpoint

**Expected behavior**
Unpublished content should not display.

**Actual behavior**
Unpublished content appears.

In Demo I created:

- Content Types
-- main bit (parent of the relationship - one to many)
---one contentlet - "Parent Item"
--other thing (child items)
---Three contentlets
----"Child One" (published)
----"Child Two" (published)
----"Child Three Unpublished" (unpublished)
-Page - "testpage"

Example code for displaying the related content:
```

#foreach($con in $dotcontent.pull("+contentType:MainBit +live:true",10,"modDate desc"))
<p>$con.title</p>
#editContentlet($con.inode)
#set($relList = $con.getrel)
    <h4>Item(s):</h4>
    <ul>
        #foreach($relatedTitle in $relList)
            <li>$relatedTitle.title</li>
        #end
    </ul>
#end
```

**Screenshots**
LIVE MODE
The contentlet "Child Three Unpublished" is in an unpublished state

![image](https://user-images.githubusercontent.com/49727631/89554631-bbadf580-d7d4-11ea-827e-70b52ca445d4.png)

WORKAROUND (from john thomas):

> What I would suggest is - instead of referencing the menu item contents as $con.menuItems, ask them to perform a formal pull of the menu items (e.g. #set( $menuItems = $dotcontent.pullRelatedField( "$con.identifier", "menuSection.menuItems", "*", "modDate desc" ) and then #foreach($item in $menuItems)), and see if they have the same behavior.  I suspect that when you pull related content directly from the relationship variable (e.g. $con.menuItems), it may not respect the Preview vs. Live vs. Edit mode.

**Comment by freddyucv** [comment]
**Note to QA:** Also fail when a contentlet has LIVE and WORKING version in this case always show the working version in LIVE mode before this change

**Comment by dsilvam** [comment]
Passed Internal QA: Working as expected. Only live items showing on Live Mode. Working Item showing in Edit and Preview Mode.

**Comment by bryanboza** [comment]
Fixed, tested on release 20.12 // Postgres // FF

**Comment by jcastro-dotcms** [comment]
The official code changes have been included in the dotCMS 5.3.8.4 LTS release only. The 5.2.8.4 LTS release is several commits behind, which made the code back-port not possible.
